### PR TITLE
Allow Kubelet to run with no Azure identity

### LINKED ISSUE #77309: Azure - Allow kubelet to run with no azure identity
Currently there is no way to run `kubelet` without any `azure` identity (be it service principal or E/MSI). In reality `azure` code running on a worker node can use `azure` via - and only via - `IMDS` without the need to access ARM. Here is the proposed change

1. establish a new mode authN = `none` this mode, is basically telling the cloud provider to work `without` any identity, this means only IMDS calls is allowed. This mode can be detected from `azure.json` if no SPN set and both MSI settings are set to `false`.
2. change [GetVolumeLimits()](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/azure_dd/azure_dd.go#L141) to use a config map within the cluster. This config map can be loaded on the cluster statically. or created - if not existent - in [Initialize()](https://github.com/kubernetes/legacy-cloud-providers/blob/master/azure/azure.go#L455)

3. Ensure that `if AuthNMode mode == none` then only `IMDS` is used  in [NodeAddress()](https://github.com/kubernetes/legacy-cloud-providers/blob/master/azure/azure_instances.go#L38) 

The net new result of the above is users bootstrapping clusters can configure the worker node to use `azure` without any identity on said worker nodes.

@feiskyer @kubernetes/sig-azure

### PR DESCRIPTION
**Pull request by feiskyer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind bug
/sig azure

**What this PR does / why we need it**:

Allow Kubelet to run with no Azure identity. useInstanceMetadata should be enabled and Kubelet would use IMDS to get node's information.

**Which issue(s) this PR fixes**:

Fixes #77309

**Special notes for your reviewer**:

#77851 is also required for no identity working.

**Does this PR introduce a user-facing change?**:

```release-note
Kubelet could be run with no Azure identity now. A sample cloud provider configure is:  `{"vmType": "vmss", "useInstanceMetadata": true, "subscriptionId": "<subscriptionId>"}`
```

### CODE REVIEW

**Comment by feiskyer** [comment]
/priority important-soon

/hold

@andyzhangx When validating this with #77851, I find AzureDisk is not working as expected. The error  message is:

```
Operation for "\"kubernetes.io/azure-disk//<pvc-resource-id>\"" failed. No retries permitted until 2019-05-15 07:37:44.323108693 +0000 UTC m=+297.614031734 (durationBeforeRetry 1m4s). Error: "MountVolume.WaitForAttach failed for volume \"pvc-dceb8b43-76e3-11e9-a1ab-000d3aa2e18b\" (UniqueName: \"kubernetes.io/azure-disk//<pvc-resource-id>\") pod \"sh\" (UID: \"dcd1c6b5-76e3-11e9-a1ab-000d3aa2e18b\") : azureDisk - WaitForAttach failed within timeout node (k8s-agentpool2-20421826-vmss000000) diskId:(k8s-vmss-dynamic-pvc-dceb8b43-76e3-11e9-a1ab-000d3aa2e18b) lun:(-1)"
```

As the error message indicated, the lun is -1, which is weird. Do you think this is related to #77483?

**Comment by feiskyer** [comment]
The above issue is fixed by #77912.

/hold cancel

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances.go [hunk 1]
```diff
@@ -316,10 +326,17 @@ func (az *Cloud) InstanceType(ctx context.Context, name types.NodeName) (string,
 		if err != nil {
 			return "", err
 		}
-		if isLocalInstance {
-			if metadata.Compute.VMSize != "" {
-				return metadata.Compute.VMSize, nil
+		if !isLocalInstance {
+			if az.vmSet != nil {
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances.go]: why do this code change? not use `metadata.Compute.VMSize, nil` directly?

Comment (feiskyer) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances.go]: when this is not local instance, and credentials not provided, we should report errors.

the original logic has been moved to L331.

**Review by neolit123 — COMMENTED** [review]
@feiskyer the change LGTM
/lgtm
/hold

but there is a good chance the release note with `{ .. }` might break anago (the release tool)
possibly trim it to?
> Kubelet could be run with no Azure identity now.

**Comment by feiskyer** [comment]
Thanks, let me remove the {} and wrap the example in quotes.

**Comment by feiskyer** [comment]
@neolit123 looked at the old release notes, `{}` is allowed within ``, so changed the example within `` now.

**Comment by neolit123** [comment]
ok, understood.
/hold cancel

**Pull request merged**